modules/plugins/youtube.go

```diff
@@ -199,93 +218,230 @@ func (yt *YouTube) actionChannel(args []string, in *discordgo.Message, out **dis
 		return yt.actionFinish
 	}
 
-	*out = yt.getChannelInfo(item.Id.ChannelId)
+	// Very few channels only have snippet.ChannelID
+	// Maybe it's youtube API bug.
+	channelId := item.Id.ChannelId
+	if channelId == "" {
+		channelId = item.Snippet.ChannelId
+	}
+	*out = yt.getChannelInfo(channelId)
+
 	return yt.actionFinish
 }
 
+// _yt channel add <channel id/link/search keywords> <discord channel>
 func (yt *YouTube) actionAddChannel(args []string, in *discordgo.Message, out **discordgo.MessageSend) youtubeAction {
 	if len(args) < 4 {
 		*out = yt.newMsg("bot.arguments.too-few")
 		return yt.actionFinish
 	}
 
-	// _yt video add <video id/link> <discord channel>
+	// check permission
+	if helpers.IsMod(in) == false {
+		*out = yt.newMsg("mod.no_permission")
+		return yt.actionFinish
+	}
+
+	// check discord channel
+	dc, err := helpers.GetChannelFromMention(in, args[len(args)-1])
+	if err != nil {
+		*out = yt.newMsg("bot.arguments.invalid")
+		return yt.actionFinish
+	}
 
-	*out = yt.newMsg("bot.arguments.invalid")
+	// search channel
+	yc, err := yt.searchQuerySingle(args[2:len(args)-1], "channel")
+	if err != nil {
+		yt.logger().Error(err)
+		*out = yt.newMsg(err.Error())
+		return yt.actionFinish
+	}
+
+	if yc == nil {
+		*out = yt.newMsg("plugins.youtube.channel-not-found")
+		return yt.actionFinish
+	}
+
+	content := DB_Youtube_Content_Channel{
+		ID:   yc.Id.ChannelId,
+		Name: yc.Snippet.ChannelTitle,
+	}
+
+	// Very few channels only have snippet.ChannelID
+	// Maybe it's youtube API bug.
+	if content.ID == "" {
+		content.ID = yc.Snippet.ChannelId
+	}
+
+	if content.ID == "" || content.Name == "" {
+		*out = yt.newMsg("plugins.youtube.channel-not-found")
+		return yt.actionFinish
+	}
+
+	// fill db entry
+	entry := DB_Youtube_Entry{
+		ServerID:                dc.GuildID,
+		ChannelID:               dc.ID,
+		NextCheckTime:           time.Now().Unix(),
+		LastSuccessfulCheckTime: time.Now().Unix(),
+
+		ContentType: "channel",
+		Content:     content,
+	}
+
+	// insert entry into the db
+	_, err = yt.createEntry(entry)
+	if err != nil {
+		yt.logger().Error(err)
+		*out = yt.newMsg(err.Error())
+		return yt.actionFinish
+	}
+
+	yt.quota.IncEntryCount()
+
+	*out = yt.newMsg("Added youtube channel `" + yc.Snippet.ChannelTitle + "` to the discord channel <#" + dc.ID + ">")
```

Comment:
Please add this to the i18n json for easier editing

Reply:
changed